### PR TITLE
Flags modifiers for add and sub

### LINKED ISSUE #21: Implement flag modifiers
An instruction can be modified with "!", for example: sub!, will change the vm flags depending on the result of the operation.

### PR DESCRIPTION
**Pull request by fkrause98**
## Changes:
- Handle modifiers for instructions, eg: `add! `should change flags depending on its result.
- Closes #21.

### CODE REVIEW

**Review by gianbelinche — APPROVED** [review]
LGTM

File: src/state.rs [hunk 1]
```diff
@@ -40,25 +43,35 @@ impl VMState {
...
     }
 
-    pub fn new_with_flag_state(flag_lt: bool, flag_eq: bool, flag_gt: bool) -> Self {
+    pub fn new_with_flag_state(flag_lt_of: bool, flag_eq: bool, flag_gt: bool) -> Self {
         let registers = [U256::zero(); 15];
         let current_frame = CallFrame::new(vec![]);
         Self {
             registers,
-            flag_lt,
+            flag_lt_of,
```
Comment (juan518munoz) [inline src/state.rs:58]: Why the rename?

Comment (fkrause98) [reply src/state.rs:58]: I originally thought this flag was only "less than", but upon re-reading the spec, I've noticed it's actually  "overflow or less than", hence the rename.

**Review by juan518munoz — APPROVED** [review]
LGTM

File: src/state.rs (hunk 1 above)
Comment (juan518munoz) [reply src/state.rs:58]: Correct, [here](https://github.com/matter-labs/zksync-era/blob/main/docs/specs/zk_evm/vm_specification/zkSync_era_virtual_machine_primer.md#vm-architecture) it states nothing about overflow, yet it is in the [specification document](https://github.com/matter-labs/zksync-era/blob/main/docs/specs/zk_evm/vm_specification/EraVM_formal_specification.pdf).

**Pull request merged**